### PR TITLE
chore(TopMenu): code scout

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Navigation menu logic code scout changes.

## Justify why they are needed

Took the opportunity while working with something else.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
* **#4359** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4359?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4358** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4358?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4357** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4357?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @guilhermespopolin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/blocks/HeaderBlock.tsx [hunk 1]
```diff
@@ -187,12 +187,10 @@ export const ProductNavContainerBlock = ({ blok, variant }: ProductNavContainerB
         value={blok.name}
         {...storyblokEditable(blok)}
       >
-        <Space y={{ base: 1.5, lg: 0 }}>
-          <NavigationTrigger href={PageLink.store({ locale })}>{blok.name}</NavigationTrigger>
-          <NavigationMenuPrimitive.Content className={navigationContent}>
-            {content}
-          </NavigationMenuPrimitive.Content>
-        </Space>
+        <NavigationTrigger href={PageLink.store({ locale })}>{blok.name}</NavigationTrigger>
```
Comment (guilhermespopolin) [inline apps/store/src/blocks/HeaderBlock.tsx:190]: The spacing intended to be added by this wrapper is being added via CSS.

File: apps/store/src/blocks/HeaderBlock.tsx [hunk 2]
```diff
@@ -266,7 +260,7 @@ export const HeaderBlock = ({ blok, ...headerProps }: HeaderBlockProps) => {
     [blok.navMenuContainer],
   )
   return (
-    <Header {...storyblokEditable(blok)} {...headerProps}>
+    <Header {...storyblokEditable(blok)}>
```
Comment (guilhermespopolin) [inline apps/store/src/blocks/HeaderBlock.tsx:263]: Before, `Header` was caring about `overlay` and `static` props. That's not the case anymore and, in any case, storyblok is not sending any props for `Header` so we don't need to spread it here.

File: apps/store/src/components/Header/Header.tsx [hunk 3]
```diff
@@ -58,7 +58,7 @@ export const Header = (props: PropsWithChildren<unknown>) => {
   }
 
   return (
-    <div style={initialStyles} className={ghostWrapper}>
+    <div className={ghostWrapper}>
```
Comment (guilhermespopolin) [inline apps/store/src/components/Header/Header.tsx:61]: Inline styles here is not necessary

File: apps/store/src/components/Header/TopMenuDesktop/TopMenuDesktop.tsx [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
 }
 
 export const TopMenuDesktop = ({ children }: TopMenuDesktopProps) => {
-  const [activeItem, setActiveItem] = React.useState('')
+  const [activeItem, setActiveItem] = useState('')
   const pathname = usePathname()
+
   useEffect(() => {
-    ;() => setActiveItem('')
+    setActiveItem('')
```
Comment (guilhermespopolin) [inline apps/store/src/components/Header/TopMenuDesktop/TopMenuDesktop.tsx:15]: That one I'd need to double check but previous version looks like a mistake added by editor auto-comple, no? 🤔

**Review by alebedev — APPROVED** [review]
Looks great! This part was pretty messy and it's much cleaner now

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Jun 12, 3:16 AM EDT**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4357).

**Pull request merged**